packages/sui-changelog/bin/index.js

```diff
@@ -0,0 +1,178 @@
+#!/usr/bin/env node
+/* eslint no-console:0 */
+const program = require('commander')
+const parseDiff = require('what-the-diff').parse
+const exec = require('child_process').exec
+const execSync = require('child_process').execSync
+const json2md = require('json2md')
+const fs = require('fs')
+const log = console.log
+const error = console.error
+const exit = process.exit
+const date = new Date()
+const {
+  DEFAULT_SCOPES,
+  PACKAGE_FILES,
+  MAX_BUFFER,
+  version,
+  scopes: customScopes = [],
+  versionRegExp,
+  oldVersionRegExp,
+  oldPackageVersionRegExp,
+  forbiddenExpressionsInCommitRegExp
+} = require('../src/config')
+const {
+  updateFileVersion,
+  getModifiedRepository,
+  getRepositoryUrl,
+  isMonoRepo,
+  hasApiError
+} = require('../src/helpers')
+
+let newPackageVersion = version
+let modifiedPackages = []
+let changelogData = []
+
+program.version(version).parse(process.argv)
+
+// Retrieve modified packages info from npm shrinkwrap.
+exec('git diff npm-shrinkwrap.json', {maxBuffer: MAX_BUFFER}, (err, stdout) => {
+  if (err) error(err)
+  const [diff = {}] = parseDiff(stdout)
+  const {hunks} = diff
+
+  if (!hunks) {
+    log('There are no changes in your shrinkwrap.')
+    exit()
+  }
+
+  // Update package versions.
+  let oldPackageVersionParts
+  hunks.find(({lines}) => {
+    return lines.find(line => {
+      oldPackageVersionParts = line.match(oldPackageVersionRegExp)
+      return oldPackageVersionParts
+    })
+  })
+
+  if (!oldPackageVersionParts)
+    PACKAGE_FILES.forEach(filePath =>
+      updateFileVersion({filePath, newPackageVersion})
+    )
+
+  changelogData.push({
+    h2: `${newPackageVersion} (${date.getDate()}/${date.getMonth() +
+      1}/${date.getFullYear()})`
+  })
+
+  log('MODIFIED PACKAGES:')
+  hunks.forEach(({lines}) => {
+    // Get modified packages by filtering lines with git removal syntax from
+    // diff.
+    const linesWithModifiedPackages = lines.filter(line =>
+      line.match(oldVersionRegExp)
+    )
+    linesWithModifiedPackages.forEach(oldVersionLine => {
+      const oldVersionIndex = lines.indexOf(oldVersionLine)
```

Comment:
you could extract this kind of callbacks with a descriptive name in order to make the code more readable

Reply:
Done!